### PR TITLE
Repeat flakiness detection performance build in the same build

### PR DESCRIPTION
**Pull request by blindpirate**
Previously we had issues in splitting performance test scenarios into buckets
with flakiness detection. In some corner case, the same scenarios can be split
into same bucket because of repetition, triggering IllegalArgumentException
"More than one scenario split for bucket". This PR fixes this issue by repeating
the scenario inside bucket.

### CODE REVIEW

File: .teamcity/src/main/kotlin/configurations/PerformanceTest.kt [hunk 1]
```diff
@@ -68,20 +69,22 @@ class PerformanceTest(
...
-                        performanceTestCommandLine(
-                            "clean ${performanceTestTaskNames.joinToString(" ") { "$it --channel %performance.channel% ${type.extraParameters}" }}",
-                            "%performance.baselines%",
-                            extraParameters,
-                            os
-                        ) +
-                            buildToolGradleParameters() +
-                            buildScanTag("PerformanceTest")
-                        ).joinToString(separator = " ")
+                repeat(if (performanceTestBuildSpec.type == PerformanceTestType.flakinessDetection) 2 else 1) {
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/configurations/PerformanceTest.kt:72]: Please capture the index here as a parameter. Below it is not clear what `it` references to and I think it will reference to the one from `gradleWrapper`.

File: .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt [hunk 2]
```diff
@@ -51,12 +51,10 @@ class StatisticsBasedPerformanceTestBucketProvider(private val model: CIBuildMod
         val scenarios = determineScenariosFor(performanceTestSpec, performanceTestConfigurations)
         val testProjectToScenarioDurations = determineScenarioTestDurations(performanceTestSpec.os, performanceTestDurations)
         val testProjectScenarioDurationsFallback = determineScenarioTestDurations(Os.LINUX, performanceTestDurations)
-        val repetitions = if (performanceTestType == PerformanceTestType.flakinessDetection) 2 else 1
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:54]: For some reason I thought we have been running everything 3 times instead of 2 times for flakiness detection. I think 2 times is fine as well.

File: .teamcity/src/main/kotlin/configurations/PerformanceTest.kt [hunk 3]
```diff
@@ -68,20 +69,22 @@ class PerformanceTest(
...
+                    gradleWrapper {
+                        name = "GRADLE_RUNNER${if (it != 0) "_2" else ""}"
+                        tasks = ""
+                        workingDir = os.perfTestWorkingDir
+                        gradleParams = (
+                            performanceTestCommandLine(
+                                "clean ${performanceTestTaskNames.joinToString(" ") { "$it --channel %performance.channel% ${type.extraParameters}" }}",
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/configurations/PerformanceTest.kt:79]: I think we should drop the `clean` for the second run, so we don't need to rebuild nor the distribution nor the performance test project.

**Comment by blindpirate** [comment]
This is how it looks like now: https://builds.gradle.org/admin/editBuildRunners.html?id=buildType:Gradle_Experimental_Check_PerformanceTestFlakinessDetectionLinuxbucket42

File: .teamcity/src/main/kotlin/configurations/PerformanceTest.kt (hunk 1 above)
Comment (blindpirate) [reply .teamcity/src/main/kotlin/configurations/PerformanceTest.kt]: `gradleWrapper` is actuall using `this`, but it's always better to use an explicit parameter. Fixed.

File: .teamcity/src/main/kotlin/configurations/PerformanceTest.kt (hunk 3 above)
Comment (blindpirate) [reply .teamcity/src/main/kotlin/configurations/PerformanceTest.kt]: done

File: .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt (hunk 2 above)
Comment (blindpirate) [reply .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:54]: Yeah, that's what I thought as well.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test and merge
